release.js

```diff
@@ -36,15 +36,31 @@ module.exports = async function ({ github, context, inputs }) {
 
   const { opticUrl, npmTag, version, id } = releaseMeta
 
+  const run = runSpawn()
   if (!pr.merged) {
-    return github.rest.repos.deleteRelease({
-      owner,
-      repo,
-      release_id: id,
-    })
+    const branchName = `release/${version}`
+    const promises = await Promise.allSettled([
+      run('git', ['push', 'origin', '--delete', branchName]),
+      github.rest.repos.deleteRelease({
+        owner,
+        repo,
+        release_id: id,
+      }),
+    ])
+
+    const errors = promises.filter(p => p.reason).map(p => p.reason.message)
+    if (errors.length) {
+      core.setFailed(
+        `Something went wrong while deleting the branch or release. \n Errors: ${errors.join(
+          '\n'
+        )}`
+      )
+    }
+
+    // Both the branch and release have been deleted, return early
```

Comment:
minor: this comment is not accurate anymore now. something may have failed, but we return nonetheless